mount.go

```diff
@@ -83,13 +85,29 @@ be interacting with the file system.`)
 		gid = uint32(flags.Gid)
 	}
 
+	// if metadata-cache:ttl-secs has been set in config-file, then
+	// switch metadata cache ttls (stat-cache-ttl and type-cache-tll)
+	// to that as that takes precedence over both stat-cache-ttl and
+	// type-cache-tll.
+	var metadataCacheTTL time.Duration
+	if mountConfig.MetadataCacheConfig.TtlInSeconds != config.TtlInSecsUnsetSentinel {
```

Comment:
Add unit tests for this logic to get metadataCacheTTL from mountConfig.MetadataCacheConfig.TtlInSeconds, StatCacheTTL and TypeCacheTTL

Reply:
Done

Reply:
done